draft-add-requirements.md

```diff
@@ -104,92 +108,137 @@ the IETF may publish, such as DNS-over-QUIC {{?I-D.ietf-dprive-dnsoquic}}.
-* During communication with the (often unencrypted) resolver (A), this resolver can claim that one or more encrypted DNS resolvers (B, C, etc) are equivalent. This is known as resolver-identified.
+There are two possible sources of designation.
 
-Network-identified is preferred since it comes from the same source of information, and removes the need to talk to the Do53 resolver
-at all. However it cannot be the sole mechanism, at least for several years, since there is a large installed base of local
-network equipment that is difficult to upgrade with new features. Hence the second mechanism must support being able to
-announce an equivalent resolver using only existing widely-deployed DNS features.
+* The local network can designate one or more encrypted DNS resolvers (B, C, etc) in addition to any Do53 resolver (A) it may offer. This is known as network-identified.
+* During communication with the (often unencrypted) resolver (A), this resolver can designate one or more encrypted DNS resolvers (B, C, etc). This is known as resolver-identified.
+
+Network-identified has the advantages that it derives from the same source of information as the network's Do53 announcement, and
+removes the need to talk to the Do53 resolver at all. However it cannot be the sole mechanism, at least for several years, since
+there is a large installed base of local network equipment that is difficult to upgrade with new features. Hence the second mechanism
+should support being able to designate resolvers using only existing widely-deployed DNS features.
 
 ## Local addressing
 
 Many networks offer a Do53 resolver on an address that is not globally meaningful, e.g. {{?RFC1918}}, link-local
-or unique local addresses. To support the discovery of Encrypted DNS in these environments, a means is needed for
-the discovery process to work from a locally-addressed Do53 resolver to an Encrypted DNS resolver that is accessible
+or unique local addresses. To support the discovery of encrypted DNS in these environments, a means is needed for
+the discovery process to work from a locally-addressed Do53 resolver to an encrypted DNS resolver that is accessible
 either at the same (local) address, or at a different global address. Both options need to be supported.
 
-# Network-identified encrypted resolvers
+## Use of designation information
+
+After the client receives designation information, it must come to a decision on whether and when to use any of the
+designated resolvers. To support this process, it would be advantageous for a solution to enable the client to validate
+the source of the assertion in some way. For example for both network-identified and resolver-identified designation, it
+may be possible to verify that the designation comes from an entity who already has full control of the client's Do53
+queries. It would be beneficial to extend this to defend against attackers that have only transient control of such queries.
```

Comment:
I don't think the verification is essential for network-identified resolver. If an attacker has the capability to spoof DHCP/RA, the validation will be of no use as the attacker can point the client to Do53 and Encrypted DNS servers managed by the attacker.

Reply:
I think an attacker who can spoof DHCP/RA "already has full control of the client's Do53 queries", because they could direct that user to any server of their choice.  Perhaps we can clarify the text.

Reply:
So would you consider that there is nothing to be gained by attempting to verify network-identified designation? If so we can limit this to resolver-identified.

Reply:
I would agree with this sentiment so long as the document is clear about "resolver-identified designation" applying to the case where the network-identified resolver itself designates an encrypted resolver. If "network-identified designation" is limited to DHCP and RA then I totally agree. (I think we all understand that and that's how I read it but just calling out this should be explicit)

Reply:
Tommy's response to Chris question is a bit unclear to me. I do see some gain in being able to validate that a given IP corresponds to the IP address of the resolver provided by my network.  In that sense the binding might the IP address has a relation with the ISP that provides my IP address and so my connectivity. This is a bit different to bind the IP of the resolver to ISP_A or ISP_B. 
I do think there is some gain to check the validation of the IP address whether Do53 or eDNS is used with that IP. Of course eDNS would be preferred for DNS transaction. 

I understand that Tommy wants that resolver-identified designation requires a encrypted resolver. I guess the reason is that the encrypted resolver is seen as a trusted point from which we move to the other resolvers. This assumes the resolver provides information that it issues. The resolver may also "proxy" information it does not issue and this information may also have some trust to the client ( DNSSEC). So it seems to me that requiring network-identified resolvers to be encrypted my not be necessary.

Reply:
Agree with Tommy. DHCP/RA is typically protected using DHCP/RA Guard but persuasive monitoring is still possible even in Enterprise networks due to internal attacks (e.g, compromised network devices, see https://tools.ietf.org/html/draft-arkko-farrell-arch-model-t-04#section-3.2.1). In Home networks, WPA2 is susceptible to vulnerabilities like attacker capable of decrypting the encrypted packets.

Reply:
Sorry I wasn't being clear Daniel. All I was trying to say is:
* confirmation is required when one resolver designates another, but
* this same confirmation is not required when DHCP/RA designates an encrypted resolver

...for the same reasons Ben and Tiru already mentioned.

I think the only reason I'm focused on networks identifying encrypted resolvers is that they can already identify Do53 servers today. That scenario doesn't need any change, or if it does, it's well out of scope here. 

However, if a network wants to identify an encrypted resolver _via_ the local forwarder, that makes it a "resolver designating resolver" scenario. The network identified the unencrypted resolver (no change to standard scenario today), then the unencrypted resolver identifies the encrypted resolver (this requires a higher confirmation bar).

Does that help, or hinder?

Reply:
Ok, I think we agree on the main point. I've just committed new wording that clarifies that it's only relevant to resolver-identified.

Reply:
ok, thanks for the clarification, we are on the same page.